File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/nl_to_sql_tool.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.


from .tool import Tool
from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class NlToSqlTool(Tool):
    """
    Derived model for Natural Language (NL) to SQL Tool
    """

    def __init__(self, **kwargs):
        """
        Initializes a new NlToSqlTool object with values from keyword arguments. The default value of the :py:attr:`~oci.aidataplatform_dp.models.NlToSqlTool.tool_type` attribute
        of this class is ``NL2SQL`` and it should not be changed.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param key:
            The value to assign to the key property of this NlToSqlTool.
        :type key: str

        :param display_name:
            The value to assign to the display_name property of this NlToSqlTool.
        :type display_name: str

        :param workspace_key:
            The value to assign to the workspace_key property of this NlToSqlTool.
        :type workspace_key: str

        :param description:
            The value to assign to the description property of this NlToSqlTool.
        :type description: str

        :param tool_type:
            The value to assign to the tool_type property of this NlToSqlTool.
            Allowed values for this property are: "RAG", "SQL", "PROMPT", "NL2SQL", "MCP", "CUSTOM", "HTTP", "REFERENCE"
        :type tool_type: str

        :param is_reference:
            The value to assign to the is_reference property of this NlToSqlTool.
        :type is_reference: bool

        :param position_x:
            The value to assign to the position_x property of this NlToSqlTool.
        :type position_x: float

        :param position_y:
            The value to assign to the position_y property of this NlToSqlTool.
        :type position_y: float

        :param properties:
            The value to assign to the properties property of this NlToSqlTool.
        :type properties: dict(str, object)

        :param time_created:
            The value to assign to the time_created property of this NlToSqlTool.
        :type time_created: datetime

        :param time_updated:
            The value to assign to the time_updated property of this NlToSqlTool.
        :type time_updated: datetime

        :param created_by:
            The value to assign to the created_by property of this NlToSqlTool.
        :type created_by: str

        :param updated_by:
            The value to assign to the updated_by property of this NlToSqlTool.
        :type updated_by: str

        :param input_schema:
            The value to assign to the input_schema property of this NlToSqlTool.
        :type input_schema: oci.aidataplatform_dp.models.NlToSqlToolInputSchema

        :param tool_config:
            The value to assign to the tool_config property of this NlToSqlTool.
        :type tool_config: oci.aidataplatform_dp.models.NlToSqlToolConfiguration

        """
        self.swagger_types = {
            'key': 'str',
            'display_name': 'str',
            'workspace_key': 'str',
            'description': 'str',
            'tool_type': 'str',
            'is_reference': 'bool',
            'position_x': 'float',
            'position_y': 'float',
            'properties': 'dict(str, object)',
            'time_created': 'datetime',
            'time_updated': 'datetime',
            'created_by': 'str',
            'updated_by': 'str',
            'input_schema': 'NlToSqlToolInputSchema',
            'tool_config': 'NlToSqlToolConfiguration'
        }

        self.attribute_map = {
            'key': 'key',
            'display_name': 'displayName',
            'workspace_key': 'workspaceKey',
            'description': 'description',
            'tool_type': 'toolType',
            'is_reference': 'isReference',
            'position_x': 'positionX',
            'position_y': 'positionY',
            'properties': 'properties',
            'time_created': 'timeCreated',
            'time_updated': 'timeUpdated',
            'created_by': 'createdBy',
            'updated_by': 'updatedBy',
            'input_schema': 'inputSchema',
            'tool_config': 'toolConfig'
        }

        self._key = None
        self._display_name = None
        self._workspace_key = None
        self._description = None
        self._tool_type = None
        self._is_reference = None
        self._position_x = None
        self._position_y = None
        self._properties = None
        self._time_created = None
        self._time_updated = None
        self._created_by = None
        self._updated_by = None
        self._input_schema = None
        self._tool_config = None
        self._tool_type = 'NL2SQL'

    @property
    def input_schema(self):
        """
        **[Required]** Gets the input_schema of this NlToSqlTool.

        :return: The input_schema of this NlToSqlTool.
        :rtype: oci.aidataplatform_dp.models.NlToSqlToolInputSchema
        """
        return self._input_schema

    @input_schema.setter
    def input_schema(self, input_schema):
        """
        Sets the input_schema of this NlToSqlTool.

        :param input_schema: The input_schema of this NlToSqlTool.
        :type: oci.aidataplatform_dp.models.NlToSqlToolInputSchema
        """
        self._input_schema = input_schema

    @property
    def tool_config(self):
        """
        **[Required]** Gets the tool_config of this NlToSqlTool.

        :return: The tool_config of this NlToSqlTool.
        :rtype: oci.aidataplatform_dp.models.NlToSqlToolConfiguration
        """
        return self._tool_config

    @tool_config.setter
    def tool_config(self, tool_config):
        """
        Sets the tool_config of this NlToSqlTool.

        :param tool_config: The tool_config of this NlToSqlTool.
        :type: oci.aidataplatform_dp.models.NlToSqlToolConfiguration
        """
        self._tool_config = tool_config

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/NlToSqlTool.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Derived model for Natural Language (NL) to SQL Tool
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=NlToSqlTool.Builder.class)
@com.fasterxml.jackson.annotation.JsonTypeInfo(use=com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, include=com.fasterxml.jackson.annotation.JsonTypeInfo.As.PROPERTY, property="toolType")

public final class NlToSqlTool extends Tool {
    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
    @com.fasterxml.jackson.annotation.JsonProperty("key")
private String key;

public Builder key(String key) {
    this.key = key;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("displayName")
private String displayName;

public Builder displayName(String displayName) {
    this.displayName = displayName;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("workspaceKey")
private String workspaceKey;

public Builder workspaceKey(String workspaceKey) {
    this.workspaceKey = workspaceKey;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;

public Builder description(String description) {
    this.description = description;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("positionX")
private Float positionX;

public Builder positionX(Float positionX) {
    this.positionX = positionX;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("positionY")
private Float positionY;

public Builder positionY(Float positionY) {
    this.positionY = positionY;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("properties")
private java.util.Map<String, Object> properties;

public Builder properties(java.util.Map<String, Object> properties) {
    this.properties = properties;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
private java.util.Date timeCreated;

public Builder timeCreated(java.util.Date timeCreated) {
    this.timeCreated = timeCreated;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
private java.util.Date timeUpdated;

public Builder timeUpdated(java.util.Date timeUpdated) {
    this.timeUpdated = timeUpdated;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("createdBy")
private String createdBy;

public Builder createdBy(String createdBy) {
    this.createdBy = createdBy;
    return this;
}
@com.fasterxml.jackson.annotation.JsonProperty("updatedBy")
private String updatedBy;

public Builder updatedBy(String updatedBy) {
    this.updatedBy = updatedBy;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("inputSchema")
private NlToSqlToolInputSchema inputSchema;



public Builder inputSchema(NlToSqlToolInputSchema inputSchema) {
    this.inputSchema = inputSchema;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("toolConfig")
private NlToSqlToolConfiguration toolConfig;



public Builder toolConfig(NlToSqlToolConfiguration toolConfig) {
    this.toolConfig = toolConfig;
    return this;
}


        public NlToSqlTool build() {
            NlToSqlTool model = new NlToSqlTool(this.key
                , this.displayName
                , this.workspaceKey
                , this.description
                , this.positionX
                , this.positionY
                , this.properties
                , this.timeCreated
                , this.timeUpdated
                , this.createdBy
                , this.updatedBy
                , this.inputSchema
                , this.toolConfig);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(NlToSqlTool model) {
                this.key(model.getKey());
    this.displayName(model.getDisplayName());
    this.workspaceKey(model.getWorkspaceKey());
    this.description(model.getDescription());
    this.positionX(model.getPositionX());
    this.positionY(model.getPositionY());
    this.properties(model.getProperties());
    this.timeCreated(model.getTimeCreated());
    this.timeUpdated(model.getTimeUpdated());
    this.createdBy(model.getCreatedBy());
    this.updatedBy(model.getUpdatedBy());
    this.inputSchema(model.getInputSchema());
    this.toolConfig(model.getToolConfig());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    
    @Deprecated
    public NlToSqlTool(String key, String displayName, String workspaceKey, String description, Float positionX, Float positionY, java.util.Map<String, Object> properties, java.util.Date timeCreated, java.util.Date timeUpdated, String createdBy, String updatedBy, NlToSqlToolInputSchema inputSchema, NlToSqlToolConfiguration toolConfig) {
    super(key, displayName, workspaceKey, description, positionX, positionY, properties, timeCreated, timeUpdated, createdBy, updatedBy);
        this.inputSchema = inputSchema;
        this.toolConfig = toolConfig;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("inputSchema")
    private final NlToSqlToolInputSchema inputSchema;

    
    public NlToSqlToolInputSchema getInputSchema() {
        return inputSchema;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("toolConfig")
    private final NlToSqlToolConfiguration toolConfig;

    
    public NlToSqlToolConfiguration getToolConfig() {
        return toolConfig;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("NlToSqlTool(");
        sb.append("super=").append(super.toString(includeByteArrayContents));
        sb.append(", inputSchema=").append(String.valueOf(this.inputSchema));
        sb.append(", toolConfig=").append(String.valueOf(this.toolConfig));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof NlToSqlTool)) {
            return false;
        }

        NlToSqlTool other = (NlToSqlTool) o;
        return java.util.Objects.equals(this.inputSchema, other.inputSchema) &&
            java.util.Objects.equals(this.toolConfig, other.toolConfig) &&
            super.equals(other);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = super.hashCode();
        result = (result * PRIME) + (this.inputSchema == null ? 43 : this.inputSchema.hashCode());
        result = (result * PRIME) + (this.toolConfig == null ? 43 : this.toolConfig.hashCode());
        return result;
    }


}
